src/wallet/mod.rs

```diff
@@ -1120,17 +1137,6 @@ where
 
             psbt_input.redeem_script = derived_descriptor.psbt_redeem_script();
             psbt_input.witness_script = derived_descriptor.psbt_witness_script();
-
-            let prev_output = input.previous_output;
-            if let Some(prev_tx) = self.database.borrow().get_raw_tx(&prev_output.txid)? {
-                if derived_descriptor.is_witness() {
-                    psbt_input.witness_utxo =
-                        Some(prev_tx.output[prev_output.vout as usize].clone());
-                }
-                if !derived_descriptor.is_witness() || builder.force_non_witness_utxo {
-                    psbt_input.non_witness_utxo = Some(prev_tx);
```

Comment:
My reading of the spec is that this **should** be here for wrapped p2sh segwit (it currently isn't):

> The transaction in network serialization format the current input spends from. This should be present for **inputs that spend non-segwit outputs** and can be present for inputs that spend segwit outputs.

so is wrapped p2sh a "segwit output" well I'm not sure but the definition of `PSBT_IN_WITNESS_UTXO` makes sure to make it explicit that wrapped are in included:

> The entire transaction output in network serialization which the current input spends from. This should only be present for inputs which spend segwit outputs, **including P2SH embedded ones.**

which would indicate that wrapped p2sh is not included in the first one and so `non-segwit output` should include wrapped p2sh....myabe?

Reply:
I agree that the spec isn't super clear, but it sounds reasonable to me to consider wrapped outputs "segwit outputs", because their signatures will commit to the previous output's value just like they would do with native segwit outputs (BIP143).

That's basically the reason why you can get away with only including the previous utxo, and not the full transaction. Some wallets (I think Trezor and maybe others) will still require the full transaction even for native segwit outputs to mitigate some types of attacks (that's why there's one flag to always explicitly include it), but I think in general it shouldn't be required.

Reply:
> I agree that the spec isn't super clear, but it sounds reasonable to me to consider wrapped outputs "segwit outputs", because their signatures will commit to the previous output's value just like they would do with native segwit outputs (BIP143).

So I guess the reasoning is that since the witness in the PSBT is a wrapped segwit witness it *must* be a segwit output if it gets spent and therefore you don't need the previous transaction. Ok.